azure-monitor-opentelemetry-distro/setup.py

```diff
@@ -1,21 +1,103 @@
 # license information.
 # --------------------------------------------------------------------------
 
+
 import os
+import re
 
-import setuptools
+from setuptools import find_packages, setup
 
-BASE_DIR = os.path.dirname(__file__)
-VERSION_FILENAME = os.path.join(
-    BASE_DIR, "src", "azure", "monitor", "opentelemetry", "distro", "version.py"
-)
-PACKAGE_INFO = {}
-with open(VERSION_FILENAME, encoding="utf-8") as f:
-    exec(f.read(), PACKAGE_INFO)
+# Change the PACKAGE_NAME only to change folder and different name
+PACKAGE_NAME = "azure-monitor-opentelemetry-distro"
+PACKAGE_PPRINT_NAME = "Azure Monitor Opentelemetry Distro"
+
+# a-b-c => a/b/c
+package_folder_path = PACKAGE_NAME.replace("-", "/")
+
+
+# azure v0.x is not compatible with this package
+# azure v0.x used to have a __version__ attribute (newer versions don't)
+try:
+    import azure
+
+    try:
+        ver = azure.__version__
+        raise Exception(
+            "This package is incompatible with azure=={}. ".format(ver)
+            + 'Uninstall it with "pip uninstall azure".'
+        )
+    except AttributeError:
+        pass
+except ImportError:
+    pass
+
+# Version extraction inspired from 'requests'
+with open(os.path.join(package_folder_path, "_version.py"), "r") as fd:
+    version = re.search(
+        r'^VERSION\s*=\s*[\'"]([^\'"]*)[\'"]', fd.read(), re.MULTILINE
+    ).group(1)
+
+if not version:
+    raise RuntimeError("Cannot find version information")
 
-setuptools.setup(
-    version=PACKAGE_INFO["VERSION"],
+setup(
+    name=PACKAGE_NAME,
+    version=version,
+    description="Microsoft {} Client Library for Python".format(
+        PACKAGE_PPRINT_NAME
+    ),
+    long_description=open("README.md", "r").read(),
+    long_description_content_type="text/markdown",
+    license="MIT License",
+    author="Microsoft Corporation",
+    author_email="ascl@microsoft.com",
+    url="https://github.com/microsoft/ApplicationInsights-Python/tree/main/azure-monitor-opentelemetry-distro",
+    classifiers=[
+        "Development Status :: 4 - Beta",
+        "Programming Language :: Python",
+        "Programming Language :: Python :: 3",
+        "Programming Language :: Python :: 3.7",
+        "Programming Language :: Python :: 3.8",
+        "Programming Language :: Python :: 3.9",
+        "Programming Language :: Python :: 3.10",
+        "Programming Language :: Python :: 3.11",
+        "License :: OSI Approved :: MIT License",
+    ],
+    zip_safe=False,
+    packages=find_packages(
+        exclude=[
+            "tests",
+            "samples",
+            # Exclude packages that will be covered by PEP420 or nspkg
+            "azure",
+            "azure.monitor",
+            "azure.monitor.opentelemetry",
+        ]
+    ),
+    include_package_data=True,
+    package_data={
+        "pytyped": ["py.typed"],
+    },
+    python_requires=">=3.7",
+    install_requires=[
+        "azure-monitor-opentelemetry-exporter>=1.0.0b10",
```

Comment:
If we're going to eliminate b8 and b9, maybe we should skip all the way to <=b11 to avoid the issue with logging moduel paths.

Reply:
Incrementing the version can probably be done in a later pr. I don't want the change to be lost in these commits without context.

Reply:
Was incrementing it from b8 to b10 intentional, then?